portfolio/src/main/webapp/task.js

```diff
@@ -85,38 +87,112 @@ function updateTaskList(newTask, lengthUnit) {
   descriptionText.innerText = newTask.description;
   cardBody.appendChild(descriptionText);
 
-  const duration = document.createElement('p');
-  duration.classList.add('card-text');
-  duration.innerText = newTask.duration;
-  cardBody.appendChild(duration);
+  // Create row for all modifiable content
+  const inputRow = document.createElement('div');
+  inputRow.classList.add('form-row');
+
+  // Create a column for duration
+  const durationColumn = document.createElement('div');
+  durationColumn.classList.add('col');
+
+  const durationLabel = document.createElement('label');
+  durationLabel.setAttribute('for', 'duration-input-' + TASK_ID_COUNTER);
+  durationLabel.innerText = 'Duration:';
+
+  const durationInput = document.createElement('input');
+  durationInput.classList.add('form-control');
+  durationInput.setAttribute('id', 'duration-input-' + TASK_ID_COUNTER);
+  if (lengthUnit == TIME_UNIT.HOURS) {
+    durationInput.setAttribute('value', newTask.duration / 60);
+  } else {
```

Comment:
Maybe check time unit minutes here using else if rather than else. And maybe you can print some error message in the else case.

Reply:
Done.